### PR TITLE
fix static property circular references during class preparation

### PR DESCRIPTION
**Pull request by SirSnyder**
I have no further details or tests for this bug fix, sorry.

### CODE REVIEW

File: src/copy.h [hunk 1]
```diff
@@ -288,7 +288,7 @@ static inline zend_function* pthreads_copy_internal_function(zend_function *func
 static zend_function* pthreads_copy_function(zend_function *function) {
 	zend_function *copy = zend_hash_index_find_ptr(&PTHREADS_ZG(resolve), (zend_ulong)function);
 	
-	if (copy) {
+	if (copy && copy->op_array.run_time_cache == NULL) {
```
Comment (tpunt) [inline src/copy.h]: This change does not look related and it also looks testable. If this is the case, can you separate this out into a separate PR (with a test) to ensure that it is definitely applied?

**Comment by tpunt** [comment]
@SirSnyder I've submitted a PR to refactor the class preparation logic (see #755). Given that this changes a fair amount of code you've written, could you please have a look over it to ensure that I haven't missed anything please? Thanks.

File: src/copy.h (hunk 1 above)
Comment (dktapps) [reply src/copy.h]: I think the bug that this change fixes was already resolved by the fix for #725. This was one of the patches I tested that resolved the bug.

Comment (tpunt) [reply src/copy.h]: Oh right, thanks for the information. It looks like it is no longer possible for the run time cache to be `NULL` with the other fix then, so this change doesn't seem needed anymore.

Comment (SirSnyder) [reply src/copy.h]: You guys are right, this part is superfluous. I will update the PR, thx.

**Comment by tpunt** [comment]
Closing this since it was superseded by the changes made in #755 (which has now been merged).

**Comment by tpunt** [comment]
I've reopened this since my attempt to refactor the static preparation code has failed. I'm still unsure whether introducing a new global for this is the best way, though. Do you see any other ways? Otherwise, can you update the PR to reflect the comments made on it, and I'll look into merging it. Thanks.

File: src/pthreads.h [hunk 2]
```diff
@@ -122,6 +122,7 @@ ZEND_BEGIN_MODULE_GLOBALS(pthreads)
 	zval  this;
 	HashTable resolve;
 	HashTable *resources;
+        zend_bool prepare_static_members;
```
Comment (tpunt) [inline src/pthreads.h]: nit: this should be tab indented

**Comment by SirSnyder** [comment]
@tpunt unfortunately I see no other solution. I've updated the PR.

**Comment by tpunt** [comment]
Merged in 8b10643, thanks.
